node/src/accountant/test_utils.rs

```diff
@@ -1236,56 +1286,171 @@ impl FailedPayableDaoFactoryMock {
+        self
+    }
+
+    pub fn insert_new_records_params(mut self, params: &Arc<Mutex<Vec<BTreeSet<Tx>>>>) -> Self {
+        self.insert_new_records_params = params.clone();
+        self
+    }
+
+    pub fn insert_new_records_result(self, result: Result<(), SentPayableDaoError>) -> Self {
+        self.insert_new_records_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn retrieve_txs_params(
         mut self,
-        payment_adjuster: PaymentAdjusterMock,
-    ) -> PayableScannerBuilder {
-        self.payment_adjuster = payment_adjuster;
+        params: &Arc<Mutex<Vec<Option<RetrieveCondition>>>>,
+    ) -> Self {
+        self.retrieve_txs_params = params.clone();
         self
     }
 
-    pub fn payment_thresholds(mut self, payment_thresholds: PaymentThresholds) -> Self {
-        self.payment_thresholds = payment_thresholds;
+    pub fn retrieve_txs_result(self, result: BTreeSet<Tx>) -> Self {
+        self.retrieve_txs_results.borrow_mut().push(result);
         self
     }
 
-    pub fn pending_payable_dao(
+    pub fn update_tx_blocks_params(
         mut self,
-        pending_payable_dao: PendingPayableDaoMock,
-    ) -> PayableScannerBuilder {
-        self.pending_payable_dao = pending_payable_dao;
+        params: &Arc<Mutex<Vec<HashMap<TxHash, TransactionBlock>>>>,
+    ) -> Self {
+        self.update_tx_blocks_params = params.clone();
         self
     }
 
-    pub fn build(self) -> PayableScanner {
-        PayableScanner::new(
-            Box::new(self.payable_dao),
-            Box::new(self.pending_payable_dao),
-            Rc::new(self.payment_thresholds),
-            Box::new(self.payment_adjuster),
-        )
+    pub fn update_tx_blocks_result(self, result: Result<(), SentPayableDaoError>) -> Self {
+        self.update_tx_blocks_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn replace_records_params(mut self, params: &Arc<Mutex<Vec<BTreeSet<Tx>>>>) -> Self {
+        self.replace_records_params = params.clone();
+        self
+    }
+
+    pub fn replace_records_result(self, result: Result<(), SentPayableDaoError>) -> Self {
+        self.replace_records_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn delete_records_params(mut self, params: &Arc<Mutex<Vec<BTreeSet<TxHash>>>>) -> Self {
+        self.delete_records_params = params.clone();
+        self
+    }
+
+    pub fn delete_records_result(self, result: Result<(), SentPayableDaoError>) -> Self {
+        self.delete_records_results.borrow_mut().push(result);
+        self
+    }
+}
+
+pub struct SentPayableDaoFactoryMock {
+    make_params: Arc<Mutex<Vec<()>>>,
+    make_results: RefCell<Vec<Box<dyn SentPayableDao>>>,
+}
+
+impl SentPayableDaoFactory for SentPayableDaoFactoryMock {
+    fn make(&self) -> Box<dyn SentPayableDao> {
+        if self.make_results.borrow().len() == 0 {
+            panic!("SentPayableDao Missing.")
+        };
```

Comment:
Why??

Unnecessary. `self.make_results.borrow_mut().remove(0)` throws a well recognizable message

Reply:
It is present in other DAOs.